integration-libs/cds/src/trending-searches/trending-searches.service.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable, OnDestroy } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { CdsEndpointsService } from '../services';
+import { CdsConfig } from '../config';
+import { BaseSiteService, WindowRef } from '@spartacus/core';
+import {
+  concatMap,
+  interval,
+  last,
+  map,
+  Observable,
+  of,
+  ReplaySubject,
+  take,
+  takeWhile,
+} from 'rxjs';
+import { SearchPhrases } from './model';
+
+const TRENDING_SEARCHES_ENDPOINT_KEY = 'searchIntelligence';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TrendingSearchesService implements OnDestroy {
+  protected httpClient = inject(HttpClient);
+  protected cdsEndpointsService = inject(CdsEndpointsService);
+  protected cdsConfig = inject(CdsConfig);
+  protected baseSiteService = inject(BaseSiteService);
+  protected winRef = inject(WindowRef);
+  protected interval: any;
+  private readonly trendingSearchesSource = new ReplaySubject<
+    SearchPhrases[]
+  >();
+  private trendingSearches$ = this.trendingSearchesSource.asObservable();
+
+  constructor() {
+    this.addTrendingSearchesListener();
+  }
+
+  private checkAvailability() {
+    return interval(250).pipe(
+      concatMap((_) => of((this.winRef.nativeWindow as any)?.Y_TRACKING)),
+      map((result) => result?.config?.cdsSiteId),
+      take(100),
+      takeWhile((val) => !val, true),
+      last()
+    );
+  }
+
+  private addTrendingSearchesListener() {
+    this.checkAvailability()
+      .pipe(take(1))
+      .subscribe((cdsSiteId) => {
+        if (cdsSiteId) {
+          const originalEndpointUrl = this.cdsEndpointsService
+            .getUrl(TRENDING_SEARCHES_ENDPOINT_KEY)
+            .replaceAll('${cdsSiteId}', cdsSiteId);
+          const httpsPrefix = `https://${this.cdsConfig.cds.tenant}-${cdsSiteId}.`;
+          const modifiedUrl = originalEndpointUrl.replace(
+            /https:\/\//g,
+            httpsPrefix
+          );
+
+          const fetchTrendingSearches = () => {
+            this.httpClient.get<any>(modifiedUrl).subscribe((data) => {
+              this.trendingSearchesSource.next(data?.searchPhrases);
+            });
+          };
+          fetchTrendingSearches();
+          this.interval = setInterval(
+            () => fetchTrendingSearches(),
+            15 * 60000
+          );
+        }
+      });
```

Comment:
Since I provided a lot of comments to one method, I'll left my suggestion above. Please double check my suggestion also, because I didn't dive too deeply into the topic.

```
this.checkAvailability().pipe(
    take(1),
    filter(Boolean),
    switchMap((cdsSiteId) => {
          const url = this.constructTrendingSearchUrl(cdsSiteId); // Check the adapter pattern used in different libraries in sparacus. Maybe it will have more sense to move it to adapter.

  return timer(0, 15 * 60000).pipe(
        switchMap(() => this.httpClient.get<any>(url).pipe(
          catchError((error) => {
            return of(null); // Not sure what is better for you null or empty array. If null, then add additional filter(Boolean) before takeUntil
          })
        ))
      );
    }),
   takeUntil(this.destroy$)
).subscribe((data) => this.trendingSearchesSource.next(data.searchPhrases));
```